application/app/controllers/download_files_controller.rb

```diff
@@ -11,14 +12,17 @@ def cancel
       return redirect_back fallback_location: root_path, alert: t('download_files.file_not_found', file_id: file_id, project_id: project_id)
     end
 
+    previous_status = file.status.to_s
     if file.status.downloading?
       command_client = Command::CommandClient.new(socket_path: ::Configuration.command_server_socket_file)
       request = Command::Request.new(command: 'download.cancel', body: {project_id: project_id, file_id: file_id})
+      log_download_file_event(file, message: 'events.download_file.cancel_started', metadata: { 'filename' => file.filename, 'previous_status' => previous_status })
```

Comment:
I would move up, before the check or remove.
the previous status here does not add any extra info, I would remove if you want to keep the start